### DISCUSSION TITLE
Pass custom variables to queries at component level

### DISCUSSION THREAD

**Discussion by ixcans** [RevereCRE/relay-nextjs#55] [Q&A] [answered]
Suppose I have multiple components in a page. Each component has it's own query with different variables. 

I would like to pass different custom variables to these queries. Given the only two ways to provide variable to queries (`query` type of requests) are via path variable `[id]` or via `VariableFromContext` at page level, I wonder how can i provide custom variables at component level? 

Happy to provide any information to assist. Thanks in advance!

**Comment by rrdelaney** [comment, marked as answer]
`relay-nextjs` only supports one page-level query; any additional queries should be done with `useLazyLoadQuery`. If you're looking to compose data for multiple components I recommend using fragments.
